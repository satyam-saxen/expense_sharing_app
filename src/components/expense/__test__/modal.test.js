import "@testing-library/jest-dom/extend-expect";
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';
import renderer from 'react-test-renderer';
import AddExpenseForm from './../modal';


global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ status: 200 }),
  })
);

beforeEach(() => {
  fetch.mockClear();
});

afterEach(cleanup);

describe("Dashboard's snapshot test", () => {
  it("Sapshot Matching of whole form", () => {
    const tree = renderer.create(<AddExpenseForm otherUsers={null} user={null}></AddExpenseForm>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Dashboard element's test", () => {

  it("Should render all texts correctly correctly", () => {
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByText(/Add New Expense/)).toBeInTheDocument();
    expect(screen.getByText('Payer')).toBeInTheDocument();
    expect(screen.getByText('Description')).toBeInTheDocument();
    expect(screen.getByText('Amount')).toBeInTheDocument();
    expect(screen.getByText('Debtors')).toBeInTheDocument();
    expect(screen.getByText('Cancel')).toBeInTheDocument();
    expect(screen.getByText('Save')).toBeInTheDocument();
  });

  it("Should render all text boxes and button correctly", () => {
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Description')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Total Amount Spent')).toBeInTheDocument();
    expect(screen.getByTestId('payer-drop-down')).toBeInTheDocument();
    expect(screen.getByTestId('debtor-drop-down')).toBeInTheDocument();
  });
});

describe("Dashboard fire events tests", () => {
  it('should check for the amount to be in double format', () => {
    const amount = 567.67;
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByPlaceholderText('Total Amount Spent').value).toBe('');
    fireEvent.change(screen.getByPlaceholderText('Total Amount Spent'), { target: { value: amount } });
    expect(parseFloat(screen.getByPlaceholderText('Total Amount Spent').value)).toBe(amount);
  });

  it('Should check for the correct rendering of description value', () => {
    const desc = 'Street Food';
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByPlaceholderText('Description').value).toBe('');
    fireEvent.change(screen.getByPlaceholderText('Description'), { target: { value: desc } });
    expect(screen.getByPlaceholderText('Description').value).toBe(desc);
  });

  test('Checks payer list drop down to be present', () => {
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByTestId('payer-drop-down').value).toBe("true");
  });

  test('Checks debtors list drop down to be present', () => {
    render(<AddExpenseForm show={true} otherUsers={null} user={null}></AddExpenseForm>);
    expect(screen.getByTestId('payer-drop-down').value).toBe("true");
  });
});