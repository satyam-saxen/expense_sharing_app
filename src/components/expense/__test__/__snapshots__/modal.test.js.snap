// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard's snapshot test Sapshot Matching of whole form 1`] = `null`;
